### PR TITLE
Create new cert-manager package

### PR DESCRIPTION
**Pull request by JoshVanL**
This PR create a new `cert-manager` package containing a `Manager`. This is used as a shared manager for creating CertificateRequests for both `tls` and `server.

This PR also re-structures the Options throughout the repo to make options more local to the services themselves.

This also changes the format of the Helm values file to be more sane and easier to follow.

This will be a breaking change to the helm chart so  must be detailed in the release notes.

Branched from #67 which should be merged first.

/assign @jakexks

### CODE REVIEW

**Review by irbekrm — CHANGES_REQUESTED** [review]
Looks great to me, I think the refactor makes sense and also the watch instead of polling - we could do the same elsewhere, i.e in cert-manager tests.

I've read through the code and experimented with deploying with the different helm options and added a couple minor suggestions + this should be rebased off master I think- happy to lgtm otherwise.

There aren't many tests for the server, certmanager and tls pacakges, but I can see that the functions there are mostly wrappers around other functionality, so perhaps that's ok.

File: pkg/certmanager/certmanager.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+		return Bundle{}, fmt.Errorf("failed to wait for CertificateRequest %s/%s to be signed: %w",
+			cr.Namespace, cr.Name, err)
+	}
+
+	log.V(2).Info("signed CertificateRequest")
+
+	// If we are not preserving CertificateRequests, delete from Kubernetes
+	if !m.opts.PreserveCertificateRequests {
+		go func() {
+			if err := m.client.Delete(ctx, cr.Name, metav1.DeleteOptions{}); err != nil {
```
[the remark this answers is not in the document]
Comment (JoshVanL) [reply pkg/certmanager/certmanager.go:123]: Thanks, good catch.

**Comment by JoshVanL** [comment]
Thanks for the review @irbekrm, should be good for another look.

**Comment by irbekrm** [comment]
Thanks for the changes- looks good to me!

/lgtm

**Comment by JoshVanL** [comment]
Pod startup flake
/retest

**Pull request merged**